lib/tasks/convert_application_yml_to_proper_types.rake

```diff
@@ -0,0 +1,64 @@
+namespace :convert_application_yml_to_proper_types do
+  desc 'Take application.yml input file and convert to proper type'
+  task :convert, [:filename] => :environment do |t, args|
+    VALUE_TYPES = ['float', 'string', 'boolean', 'json', 'timestamp', 'integer']
+    filename = args[:filename]
+
+    if filename
+      content = ''
+      add_spaces = false
+      File.open(filename+'.tmp', 'w') do |newfile|
+        IO.foreach(filename) do |l|
+          add_spaces = true if l.include?('development:')
+          content=l
+          key_value = l.split(':')
+
+          if key_value.length == 2 && !key_value[1].strip.blank?
+            key = key_value[0].strip
+            value = key_value[1].strip
+            value = value.strip
+            value_type = find_value_type(key)
+            converted_val = convert(value, value_type)
+            content = "#{key}: #{converted_val}"
+            content = "  #{key}: #{converted_val}" if add_spaces
+          end
+
+          newfile.puts content
+        end
+      end
+      FileUtils.mv filename+'.tmp', filename
+    else
+      puts 'Please specify argument of filename.'
+      puts 'Example: rake convert_application_yml_to_proper_types:convert[application.yml]'
+    end
+  end
+
+  def find_value_type(key)
+    IO.foreach("#{Rails.root}/lib/identity_config.rb") do |l|
+      if l.include?(key)
+        VALUE_TYPES.each_with_index { |word, index|
+           return VALUE_TYPES[index] if l.index(word)
+        }
+      end
+
+      nil
+    end
+  end
+
+  def convert(value, value_type)
+    case value_type
+    when 'float'
+      value.tr("'", '').to_f
+    when 'boolean'
+      value=='true'
```

Comment:
because this is manual parsing... `value` here is likely going to have quotes in it, like `"true"` or `'true'` so I think you'd need to do something more like this:
```suggestion
      value.match?(/^['"]?true['"]$/)
```

Reply:
`JSON.parse` will get mad at single quotes we you could try YAML loading this as well:

```suggestion
      YAML.load(value).to_s == 'true'
```
and this will handle booleans and strings

```
YAML.load('true').to_s == 'true'
# => true
YAML.load(%|"true"|).to_s == 'true'
# => true
YAML.load(%|'true'|).to_s == 'true'
# => true
```

Reply:
I did this, but had to do JSON.parse. Sometimes it's "true", and other times it's "'true'" even after YAML.load.